### ISSUE TITLE
GlossarySection component heading should be h2

### ISSUE THREAD

**Issue by katydecorah** [mapbox/dr-ui#373]
The GlossarySection component heading is an h3. Since this component is the only content on the Glossary Help page, it should use an h2 to keep the heading hierarchy of the page in order.

https://github.com/mapbox/dr-ui/blob/e20d60a22541c9c655ff12ae33965064c4f0988b/src/components/glossary-section/glossary-section.js#L22
